sphinx/config.py

```diff
@@ -365,6 +365,37 @@ def filter(self, rebuild):
         return (value for value in self if value.rebuild in rebuild)  # type: ignore
 
 
+def eval_config_file(confdir, filename, tags):
+    # type: (unicode, unicode, Tags) -> Dict[unicode, Any]
+    """Evaluate a config file."""
+    if confdir is None:
```

Comment:
This doesn't belong here. This is a side-effect of how `sphinx-build` is implemented, whereby not passing `confdir` indicates that we're loading configuration from the command line instead (the `-C` option). Given that it's not possible to load configuration _without_ a config file, we can remove this from here and make `confdir` a required parameter.

Reply:
I added this to make `Config.__init__()` compatible with old class. But, as you said, it is not related with this function. So I moved this if-block to `Config.__init__()`.
Thanks,